### PR TITLE
Update RBAC rules for external provisioner and attacher to the updated roles required based on upstream repositories

### PR DESCRIPTION
**Pull request by davidz627**
Resolves: #281 

Should fix the "latest" test failures.

/assign @msau42 @verult

### CODE REVIEW

**Comment by verult** [comment]
/lgtm

In the future it would be nice to include snapshotter RBAC rules only for the alpha deployment, if possible

**Comment by davidz627** [comment]
moved snapshot rules to alpha overlay

File: deploy/kubernetes/overlays/alpha/rbac_add_snapshotter.yaml [hunk 1]
```diff
@@ -4,27 +4,33 @@ kind: ClusterRole
...
+    resources: ["persistentvolumeclaims"]
+    verbs: ["get", "list", "watch", "update"]
+  - apiGroups: ["storage.k8s.io"]
+    resources: ["storageclasses"]
+    verbs: ["get", "list", "watch"]
+  - apiGroups: [""]
+    resources: ["events"]
+    verbs: ["list", "watch", "create", "update", "patch"]
+  - apiGroups: [""]
+    resources: ["secrets"]
```
Comment (verult) [inline deploy/kubernetes/overlays/alpha/rbac_add_snapshotter.yaml]: Secrets were originally included in attacher roles and then later removed. Do we need the secrets permissions here for different reasons?

@jingxu97

Comment (msau42) [reply deploy/kubernetes/overlays/alpha/rbac_add_snapshotter.yaml]: I don't think so. Secrets are only required if the plugin needs them, and pd doesn't.

Comment (davidz627) [reply deploy/kubernetes/overlays/alpha/rbac_add_snapshotter.yaml]: resolved

**Pull request merged**